### DISCUSSION TITLE
bugs on iap review_notes ?

### DISCUSSION THREAD

**Discussion by schunlee** [fastlane/fastlane#17671] [General]
`      

        Spaceship::Tunes.login($FASTLANE_USER, $FASTLANE_PASSWORD)
        app = Spaceship::Application.find(ENV['APP_IDENTIFIER'])
        iaps = app.in_app_purchases.all

        for i in 0..iaps.size - 1
            if iaps.at(i).product_id == $PRODUCT_ID
                target_iap = iaps.at(i).edit
                puts target_iap`

> 	review_notes=nil, 

I see some bugs on Spaceship, here the review_notes of target_iap will be always nil, actually before I already input some sentences on Review Notes via web browser. Now using Spaceship, I just want to uodate Cleared for Sale checkbox, but I also set review notes blank.

**Comment by dinsen** [comment]
If I understand the code right (I'm still learning the codebase), it has never worked since it's not being mapped here:
https://github.com/fastlane/fastlane/blob/master/spaceship/lib/spaceship/tunes/iap_detail.rb#L49-L57

**Reply by dinsen** [reply]
Ohh it's not the mapping, but it needs a method to expose itself like reivew_screenshots
https://github.com/fastlane/fastlane/blob/master/spaceship/lib/spaceship/tunes/iap_detail.rb#L158-L162

**Comment by rogerluan** [comment]
Hey @schunlee 👋 

If I were to read them manually like you're doing, I'd use the same approach used in `Precheck`, that it's proven to work. You can check its implementation here: https://github.com/fastlane/fastlane/blob/master/precheck/lib/precheck/rule_processor.rb#L212-L241

One basic difference I noticed is that you're accessing via `app.in_app_purchases` and Precheck uses `Spaceship::Tunes.client.iaps(app_id: app_id)`. Notice that IAP doesn't support API key (it doesn't look like you're using API key, but just a friendly reminder 🤗 )

Good luck!
